ironic_prometheus_exporter/messaging.py

```diff
@@ -0,0 +1,30 @@
+import os
+
+from oslo_config import cfg
+from oslo_messaging.notify import notifier
+
+prometheus_opts = [
+    cfg.StrOpt('file_name', required=True),
+    cfg.StrOpt('file_dir', required=True)
+]
+
+
+def register_opts(conf):
+    conf.register_opts(prometheus_opts, group='oslo_messaging_notifications')
+
+
+class PrometheusFileDriver(notifier.Driver):
+
+    "Publish notifications into a File to be used by Prometheus"
+
+    def __init__(self, conf, topics, transport):
+        self.file_dir = conf.oslo_messaging_notifications.file_dir
+        self.file_name = conf.oslo_messaging_notifications.file_name
+        if not os.path.exists(self.file_dir):
+            os.makedirs(self.file_dir)
+        super(PrometheusFileDriver, self).__init__(conf, topics, transport)
+
+    def notify(self, ctxt, message, priority, retry):
+        prometheus_file = open((self.file_dir + '/' + self.file_name), 'w')
+        prometheus_file.write(str(message))
```

Comment:
It doesn't look like this is going to produce the right file format. Isn't the point of this plugin to generate the prometheus formatted data to be served by something else?

Reply:
Hi Doug, if I do remember we want the driver as simple as possible, we will format the data in the file after is written

Reply:
That's  not what I remember from the conversation at all. The idea was to have the notifier emit the data in a format that is ready to be served, and then provide a very simple web service to allow prometheus to ask for the data.
If you're going to store the message in some format for later processing, you need to use JSON or some other encoding system in order to have parseable data.

Reply:
My recollections mathes Doug's. I thought this patch was just the first step and the next patches would introduce the correct format.

Reply:
So Doug and Dmitry's recollections don't match Iury's and my recollections.

The orignally discussed point, and as written in the stories this work is based upon is to a) collect the data, and once we have all of it, whatever is exposing that data to prometheus's web browser (because prometheus may not visit that often as the metrics are updated), we then collect all of the files and transform it to a prometheus friendly format before handing back over the webserver socket. I think I view the filtering as a heavy-ish load but still a very simple thing because it doesn't need to be super featureful, it just needs to iterate through data and return what it has. Where as this is going to be writing files per node to some folder with no solid consistency between drivers so for each driver we're going to have to add transformations as we go.

Reply:
So I should keep the driver simple just writing the messages into a json file? What you everyone think about it? I have no problem to change to do the parser for the messages too, but this would require some samples of the message format etc.

Reply:
I think it would be best, but we could just do it all in the plugin. :\

To get the sample data, I think we should use this plugin to dump the data out on one of the dev machines we have. To do so, we should loop in @derekhiggins. Worst comes to worst, I can power on my supermicro machine in my house and get it back online, although the message data will differ being a different vendor than our initial target.

Reply:
So maybe we can go with the following workflow:
1- update the plugin here to write to a json file
2- Try to use the plugin in a real env so we can collect the data (Only one node is fine for this, but I would say that two would make thinks easier since we would be able to map correctly for N nodes)
3- Update the plugin to parse the data 
What everyone think about this?

Reply:
That plan works for me.

Reply:
I think this path brings us forward faster, and from a transformation standpoint, we could likely take a list of the objects we want and just being transforming them as we go or dropping them on the floor.

Reply:
Thanks everyone, changes have been made =)

Reply:
after talking to @iurygregory, he is going to use the plugin to gather real metrics once the initial version of the plugin merged.